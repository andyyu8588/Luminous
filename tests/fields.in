class A {
	private a;
	protected b;
	public c;
	
	public constructor() {
		this.a = "A's a";
		this.b = "A's b";
		this.c = "A's c";
	}

	public printA() {
		print(this.a);
	}

    public printB() {
        print(this.b);
    }
}

class B inherits A {
    private d;
    public e;

    public constructor(d) {
        this.b = "B's b";
        this.c = "B's c";
        this.d = d;
    }

    public printD() {
        print(this.d);
    }

    public printB() {
        print(this.b);
    }
}

instanceA = A();
instanceB = B("d");
instanceA.printA();
instanceA.printB();
instanceB.printB();
print(instanceA.c);
print(instanceB.c);
instanceB.printD();
print(instanceB.e);

