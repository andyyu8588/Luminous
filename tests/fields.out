A's a
A's b
B's b
A's c
B's c
d
null
